packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_create_loan_scheme.test.ts

```diff
@@ -0,0 +1,137 @@
+import { DeFiDRpcError, GenesisKeys } from '@defichain/testcontainers'
+import { getProviders, MockProviders } from '../provider.mock'
+import { P2WPKHTransactionBuilder } from '../../src'
+import { fundEllipticPair, sendTransaction } from '../test.utils'
+import { WIF } from '@defichain/jellyfish-crypto'
+import { BigNumber } from '@defichain/jellyfish-json'
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { Testing } from '@defichain/jellyfish-testing'
+
+const container = new LoanMasterNodeRegTestContainer()
+const testing = Testing.create(container)
+
+let providers: MockProviders
+let builder: P2WPKHTransactionBuilder
+
+beforeAll(async () => {
+  await testing.container.start()
+  await testing.container.waitForWalletCoinbaseMaturity()
+
+  providers = await getProviders(testing.container)
+  providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[GenesisKeys.length - 1].owner.privKey))
+  builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic)
+
+  // NOTE(jingyi2811): default scheme
+  await testing.container.call('createloanscheme', [100, new BigNumber(1.5), 'default'])
+  await testing.generate(1)
+})
+
+afterAll(async () => {
+  await testing.container.stop()
+})
+
+beforeEach(async () => {
+  // Fund 10 DFI UTXO
+  await fundEllipticPair(testing.container, providers.ellipticPair, 10)
+  await providers.setupMocks() // required to move utxos
+})
+
+afterEach(async () => {
+  const result = await testing.container.call('listloanschemes')
+  const data = result.filter((r: { default: boolean }) => !r.default)
+
+  for (let i = 0; i < data.length; i += 1) {
+    // NOTE(jingyi2811): Delete all schemes except default scheme
+    await testing.container.call('destroyloanscheme', [data[i].id])
+    await testing.generate(1)
+  }
+})
+
+describe('loan.createLoanScheme()', () => {
```

Comment:
beforeAll, afterAll, beforeEach and afterEach should be in describe block's scope as in other tests

Reply:
I compare to other files in the same directory. They are like this.

Reply:
Yes I see that now.
The logic is to reduce global variables to a strict minimum. In that case it does not do much harm but it will be better in the long run.